ClosedXML/Excel/CalcEngine/Functions/Text.cs

```diff
@@ -275,6 +276,61 @@ private static object _Text(List<Expression> p)
             else
                 return nf.Format(number, CultureInfo.InvariantCulture);
         }
+        /// <summary>
+        /// A function to Join text https://support.office.com/en-us/article/textjoin-function-357b449a-ec91-49d0-80c3-0e8fc845691c
+        /// </summary>
+        /// <param name="p">Parameters</param>
+        /// <returns> string </returns>
+        /// <exception cref="ApplicationException">
+        /// Delimiter in first param must be a string
+        /// or
+        /// Second param must be a boolen (TRUE/FALSE)
+        /// </exception>
+        private static object TextJoin(List<Expression> p)
+        {
+            List<string> values = new List<string>();
+            var p0 = p[0].Evaluate();
+            var p1 = p[1].Evaluate();
+
+            if (!(p0 is string)) throw new ApplicationException("Delimiter in first param must be a string");
+            var delimiter = (string) p0;
+
+            if (!(p1 is bool)) throw new ApplicationException("Second param must be a boolen (TRUE/FALSE)");
+            var ignoreEmptyStrings = (bool) p1;
+
+
+            foreach(Expression param in p.Skip(2))
```

Comment:
Please fix whitespace formatting. We use the CodeMaid Visual Studio plugin. Run CodeMaid Clean on all affected files.

Reply:
Use implicit variable type declaration, i.e. `foreach (var param in p.Skip(2))`